### PR TITLE
Optionally return p-value from param_estimates()

### PR DESCRIPTION
**Pull request by barrettk**
closes https://github.com/metrumresearchgroup/bbr/issues/458

### CODE REVIEW

**Comment by barrettk** [comment]
Considered adding this functionality to functions like `param_estimates_batch()` and `param_estimates_compare()`, but after inspecting the returned dataframes and required input, I dont think it makes sense to append it to those functions

**Comment by seth127** [comment]
@kyleam would you mind reviewing this?

Also, @callistosp take a look at this if you get a second to make sure it's what you had in mind for #458 and that it works how you'd expected.

**Review by kyleam — CHANGES_REQUESTED** [review]

File: R/param-estimates.R [hunk 1]
```diff
@@ -95,6 +98,18 @@ param_estimates.bbi_nonmem_summary <- function(.summary) {
 
   param_df <- add_param_shrinkage(param_df, .summary)
 
+  # Add p-value
+  if(!is.null(.alpha)){
+    assert_numeric(.alpha)
+    pval_df <-
+      param_df %>%
+      filter(str_detect(parameter_names, "OMEGA") & diag == TRUE) %>%
+      bind_cols(data.frame(pval=.summary$shrinkage_details[length(.summary$shrinkage_details)][[1]][[1]]$pval)) %>%
```
[the remark this answers is not in the document]
Comment (barrettk) [reply R/param-estimates.R:107]: @kyleam I do like this idea. Only issue is running that command multiple times causes duplicate columns to appear:
image

I still like nesting the functionality within `add_param_shrinkage` though, so will address this

[the remark this answers is not in the document]
Comment (barrettk) [reply R/param-estimates.R:107]: nevermind I take that back, didnt see that you removed the previous call!

**Comment by barrettk** [comment]
@kyleam how do you feel about the test? It looks like we arent testing `add_param_shrinkage` since its an internal function. I think the integration test (not sure if thats correct terminology in this case) I added should be sufficient, but not 100% confident on that

File: R/param-estimates.R [hunk 2]
```diff
@@ -44,8 +46,10 @@ param_estimates <- function(.summary) {
 #' @importFrom purrr map_at map_depth map_lgl
 #' @importFrom rlang list2
 #' @importFrom stringr str_detect
+#' @importFrom checkmate assert_numeric
+#' @importFrom dplyr bind_cols
```
Comment (kyleam) [inline R/param-estimates.R:50]: `bind_cols` is no longer used.

**Review by kyleam — APPROVED** [review]
@barrettk Thanks, this looks good to me.

Left one minor comment about an unused import.

**Comment by kyleam** [comment]
> @kyleam how do you feel about the test? [...]

I agree, I don't see any value to testing `add_param_shrinkage` directly.

File: R/param-estimates.R [hunk 3]
```diff
@@ -95,6 +98,18 @@ param_estimates.bbi_nonmem_summary <- function(.summary) {
 
   param_df <- add_param_shrinkage(param_df, .summary)
 
+  # Add p-value
+  if(!is.null(.alpha)){
+    assert_numeric(.alpha)
```
[the remark this answers is not in the document]
Comment (kyleam) [reply R/param-estimates.R]: > It did show up as already imported on my end, but ill add another call just to be safe

Right, two functions in `cov-cor.R` already have `@importFrom checkmate assert_numeric`, which why this doesn't trigger an error.

My view is that a function shouldn't rely on `@importFrom`s from other functions (i.e. each function's `@importFrom`s should give a complete list of what imports it needs).  That way if someone modifies the `cov-cor.R` functions in a way that drops an import, they don't have to search around the code base for some other function that is implicitly depending on the import.

**Comment by barrettk** [comment]
@kyleam unfortunately, making the fix automatically removed your review. Can you approve again?

**Review by kyleam — APPROVED** [review]

**Pull request merged**